cray_product_catalog/catalog_delete.py

```diff
@@ -61,19 +61,153 @@
+    """
+
+    def __init__(self):
+        self.__main_cm = None
+        self.__product_cm = None
+        self.__cm_namespace = None
+        self.__product_name = None
+        self.__product_version = None
+        self.__max_reties_for_main_cm = None
+        self.__max_reties_for_prod_cm = None
+        self.__key = None
+        self.__main_cm_fields = None
+        self.__product_cm_fields = None
+
+    # property definitions
+    @property
+    def main_cm(self):
+        return self.__main_cm
+
+    @main_cm.setter
+    def main_cm(self, __main_cm):
+        self.__main_cm = __main_cm
+
+    @property
+    def product_cm(self):
+        return self.__product_cm
+
+    @product_cm.setter
+    def product_cm(self, __product_cm):
+        self.__product_cm = __product_cm
+
+    @property
+    def cm_namespace(self):
+        return self.__cm_namespace
+
+    @cm_namespace.setter
+    def cm_namespace(self, __cm_namespace):
+        self.__cm_namespace = __cm_namespace
+
+    @property
+    def product_name(self):
+        return self.__product_name
+
+    @product_name.setter
+    def product_name(self, __product_name):
+        self.__product_name = __product_name
+
+    @property
+    def product_version(self):
+        return self.__product_version
+
+    @product_version.setter
+    def product_version(self, __product_version):
+        self.__product_version = __product_version
+
+    @property
+    def max_reties_for_main_cm(self):
+        return self.__max_reties_for_main_cm
+
+    @max_reties_for_main_cm.setter
+    def max_reties_for_main_cm(self, __max_reties_for_main_cm):
+        self.__max_reties_for_main_cm = __max_reties_for_main_cm
+
+    @property
+    def max_reties_for_prod_cm(self):
+        return self.__max_reties_for_prod_cm
+
+    @max_reties_for_prod_cm.setter
+    def max_reties_for_prod_cm(self, __max_reties_for_prod_cm):
+        self.__max_reties_for_prod_cm = __max_reties_for_prod_cm
+
+    @property
+    def key(self):
+        return self.__key
+
+    @key.setter
+    def key(self, __key):
+        self.__key = __key
+
+    @property
+    def main_cm_fields(self):
+        return self.__main_cm_fields
+
+    @main_cm_fields.setter
+    def main_cm_fields(self, __main_cm_fields):
+        self.__main_cm_fields = __main_cm_fields
+
+    @property
+    def product_cm_fields(self):
+        return self.__product_cm_fields
+
+    @product_cm_fields.setter
+    def product_cm_fields(self, __product_cm_fields):
+        self.__product_cm_fields = __product_cm_fields
+
+    # private methods
+    def __key_belongs_to_main_cm_fields(self):
+        return self.__key in self.__main_cm_fields
+
+    def __key_belongs_to_prod_cm_fields(self):
+        return self.__key in self.__product_cm_fields
+
+    def __modify_main_cm(self):
+        LOGGER.info(f"Removing from config_map={self.main_cm} in namespace={self.cm_namespace} "
+                    f"for {self.product_name}/{self.product_version} (key={self.key})")
```

Comment:
LOGGER calls should use lazy string evaluation rather than f-strings
```suggestion
        LOGGER.info("Removing from ConfigMap=%s in namespace=%s for %s/%s (key=%s)",
                    self.main_cm, self.cm_namespace, self.product_name, self.product_version, self.key)
```

Reply:
Modified as stated